packages/web3-eth/test/integration/batch.test.ts

```diff
@@ -0,0 +1,79 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import WebSocketProvider from 'web3-providers-ws';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { hexToNumber } from 'web3-utils';
+import { Web3Eth } from '../../src';
+
+import { createNewAccount, getSystemTestProvider, isWs } from '../fixtures/system_test_utils';
+
+describe('eth', () => {
+	let web3Eth: Web3Eth;
+	let accounts: string[] = [];
+	let clientUrl: string;
+
+	beforeAll(async () => {
+		clientUrl = getSystemTestProvider();
+		const acc1 = await createNewAccount({ unlock: true, refill: true });
+		const acc2 = await createNewAccount({ unlock: true, refill: true });
+		accounts = [acc1.address, acc2.address];
+		if (isWs) {
+			web3Eth = new Web3Eth(
+				new WebSocketProvider(
+					clientUrl,
+					{},
+					{ delay: 1, autoReconnect: false, maxAttempts: 1 },
+				),
+			);
+		} else {
+			web3Eth = new Web3Eth(clientUrl);
+		}
+	});
+	afterAll(() => {
+		if (isWs && web3Eth?.provider) {
+			(web3Eth.provider as WebSocketProvider).disconnect();
+		}
+	});
+
+	describe('methods', () => {
+		it('BatchRequest', async () => {
+			const batch = new web3Eth.BatchRequest();
+			const request1 = {
+				id: 10,
+				method: 'eth_getBalance',
+				params: [accounts[0], 'latest'],
+			};
+			const request2 = {
+				id: 11,
+				method: 'eth_getBalance',
+				params: [accounts[1], 'latest'],
+			};
+			batch.add(request1).catch(console.error);
+			batch.add(request2).catch(console.error);
```

Comment:
We also need to make sure each request was resolved successfully.

Reply:
I've checked it on next line
```ts 
const [response1, response2] = await batch.execute();
expect(response1.result).toBeDefined()
expect(response2.result).toBeDefined();
```